src/tlo/methods/cervical_cancer.py

```diff
@@ -0,0 +1,1802 @@
+"""
+Cervical Cancer Disease Module
+
+Limitations to note:
+At some point we may need to specify the treatment eg total hysterectomy plus or minus chemotherapy
+but we agree not now
+Consider in future making hpv acquisition risk depend on current prevalence of hpv
+
+"""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, List
+
+import numpy as np
+import pandas as pd
+
+from tlo import DateOffset, Module, Parameter, Property, Types, logging
+from tlo.events import IndividualScopeEventMixin, PopulationScopeEventMixin, RegularEvent
+from tlo.lm import LinearModel, LinearModelType, Predictor
+from tlo.methods import Metadata
+from tlo.methods.cancer_consumables import get_consumable_item_codes_cancers
+from tlo.methods.causes import Cause
+from tlo.methods.demography import InstantaneousDeath
+from tlo.methods.dxmanager import DxTest
+from tlo.methods.healthsystem import HSI_Event
+from tlo.methods.symptommanager import Symptom
+from tlo.util import read_csv_files
+
+if TYPE_CHECKING:
+    from tlo.methods.hsi_generic_first_appts import HSIEventScheduler
+    from tlo.population import IndividualProperties
+
+from tlo.methods.hsi_generic_first_appts import GenericFirstAppointmentsMixin
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+class CervicalCancer(Module, GenericFirstAppointmentsMixin):
+    """Cervical Cancer Disease Module"""
+
+    def __init__(self, name=None, resourcefilepath=None):
+        super().__init__(name)
+        self.resourcefilepath = resourcefilepath
+        self.linear_models_for_progression_of_hpv_cc_status = dict()
+        self.lm_onset_vaginal_bleeding = None
+        self.daly_wts = dict()
+        self.item_codes_cervical_can = dict()
+
+    INIT_DEPENDENCIES = {
+        'Demography', 'SimplifiedBirths', 'HealthSystem', 'Lifestyle', 'SymptomManager', 'Hiv'
+    }
+
+    OPTIONAL_INIT_DEPENDENCIES = {'HealthBurden', 'HealthSeekingBehaviour'}
+
+    ADDITIONAL_DEPENDENCIES = {'Tb'}
+
+    METADATA = {
+        Metadata.DISEASE_MODULE,
+        Metadata.USES_SYMPTOMMANAGER,
+        Metadata.USES_HEALTHSYSTEM,
+        Metadata.USES_HEALTHBURDEN
+    }
+
+    # Declare Causes of Death
+    CAUSES_OF_DEATH = {
+        'CervicalCancer': Cause(gbd_causes='Cervical cancer', label='Cancer (Cervix)'),
+    }
+
+    # Declare Causes of Disability
+    CAUSES_OF_DISABILITY = {
+        'CervicalCancer': Cause(gbd_causes='Cervical cancer', label='Cancer (Cervix)'),
+    }
+
+    PARAMETERS = {
+        "hpv_cin_options": Parameter(
```

Comment:
I wonder whether it makes sense to soft-code these? Also the parameter values combine the HPV/CIN/STAGES into one concatened list, so is it wise to seperate them here?

Reply:
deleting these as parameters